server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -152,11 +153,14 @@ public void onFailure(String file) {
                 // Track upload failure
                 segmentTracker.addUploadBytesFailed(latestFileNameSizeOnLocalMap.get(file));
             }
-        }, remoteDirectory, storeDirectory, this::getChecksumOfLocalFile);
+        }, remoteDirectory, storeDirectory);
+        this.segmentFilesInLatestMetadata = new HashSet<>();
```

Comment:
`RemoteSegmentStoreDirectory.getSegmentsUploadedToRemoteStore` should provide this info. Do we really need a new map?

Reply:
The concerned method has details about the segment files that have been uploaded successfully, but not the metadata file. It is possible that the metadata file upload has failed, but the segment files themselves can be uploaded. Also, given that we have common metadata file name for the entire generation when there can be different many refreshes, we can conclusively say anything if we know the latest uploaded metadata file name. We would need the details on what files were uploaded as part of the latest metadata file upload to ascertain that the metadata file already has been uploaded with the same set of segment files.